internal/rbd/rbd_util.go

```diff
@@ -439,10 +439,11 @@ func (rv *rbdVolume) isInUse() (bool, error) {
 	// because we opened the image, there is at least one watcher
 	defaultWatchers := 1
 	if rv.Primary {
-		// a watcher will be added by the rbd mirror daemon if the image is primary
+		// if rbd mirror daemon is running, a watcher will be added by the rbd
+		// mirror daemon for mirrored images.
```

Comment:
Looks good, this should be an improvement.

However, I wonder of there is an issue when mirroring has not been started (or is currently active), an other rbd client could connect to the image. Should this be something to look into?

Reply:
Yes just had the same discussion with @ShyamsundarR, It's difficult to identify the watcher is added by rbd mirror daemon or not. this issue exists only for RBD mirrored PVC and the watcher is extra protection from the cephcsi side or CO.

Reply:
Other than this observation LGTM!